l10n_nl_xaf_auditfile_export/auditfile.py

```diff
@@ -0,0 +1,641 @@
+        'element_name': element_name, 'value': value}
+
+
+def writeln_element(
+        outfile, element_name, value, indent=0, max_length=0, optional=False):
+    if optional and not value:
+        return ''
+    if indent:
+        outfile.write(' ' * indent)
+    outfile.write(get_element(element_name, value, max_length=max_length))
+    outfile.write('\n')
+
+
+def writeln_element_xml(outfile, element_name, value, indent=0):
+    """Use this for data that is known to be already xml."""
+    outfile.write(
+        "%(indent)s<%(element_name)s>%(value)s</%(element_name)s>\n" % {
+            'indent': ' ' * indent,
+            'element_name': element_name,
+            'value': value})
+
+
+def writeln_element_open(outfile, element_name, indent=0):
+    if indent:
+        outfile.write(' ' * indent)
+    outfile.write('<%s>\n' % element_name)
+
+
+def writeln_element_close(outfile, element_name, indent=0):
+    if indent:
+        outfile.write(' ' * indent)
+    outfile.write('</%s>\n' % element_name)
+
+
+def writeln_element_empty(outfile, element_name, indent=0):
+    if indent:
+        outfile.write(' ' * indent)
+    outfile.write('<%s />\n' % element_name)
+
+
+def write_data(outfile, data, indent=0):
+    """data contains an array of tuples that must be written as
+    elements.
+    Each tuple contains at least an element_name.
+    If the tuple has a second element it is the value.
+    If the tuple has a third element, it is the max_length of the value.
+    """
+    for element in data:
+        element_name = element[0]
+        value = len(element) > 1 and element[1] or ''
+        max_length = len(element) > 2 and element[2] or 0
+        writeln_element(
+            outfile, element_name, value, indent=indent,
+            max_length=max_length)
+
+
+def write_header(specification, outfile, connection):
+    outfile.write('<?xml version="1.0" encoding="UTF-8"?>\n')
+    outfile.write(
+        '<auditfile'
+        ' xmlns="http://www.auditfiles.nl/XAF/3.2"'
+        ' xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">\n')
+    writeln_element_open(outfile, 'header', indent=2)
+    with connection.cursor(
+            cursor_factory=psycopg2.extras.DictCursor) as cursor:
+        cursor.execute(SQL_HEADER, specification)
+        row = cursor.fetchone()
+        data = (
+            ('fiscalYear', row['y_code'], 9),
+            ('startDate', specification['date_start']),
+            ('endDate', specification['date_stop']),
+            ('curCode', row['m_name']),
+            ('dateCreated', row['date_generated']),
+            ('softwareDesc', 'Odoo'),
+            ('softwareVersion', '8.0'))
+        write_data(outfile, data, indent=4)
+    writeln_element_close(outfile, 'header', indent=2)
+
+
+def patch_country_code(country_code):
+    """Not all countrycodes are already in the xsd schema.
+
+    JE - Jersey
+    ME - Montenegro
+    """
+    if country_code in ['JE', 'ME']:
```

Comment:
That is a little sneaky, hiding an unrelated change like this in a big PR. Did you change any thing else functionally?

Reply:
I created https://github.com/OCA/l10n-netherlands/pull/166/files earlier today which addresses the same problem.